### ISSUE TITLE
Possible issue with `prefer-object-from-entries`

### ISSUE THREAD

**Issue by noftaly** [sindresorhus/eslint-plugin-unicorn#1495]
I just updated unicorn, and enabled `prefer-object-from-entries`. It reported this code as invalid:
```ts
const categories = stores
  .reduce((acc, val) => {
    const category = getCategory(val);
    if (acc[category])
      acc[category].push(val);
    else
      acc[category] = [val];
    return acc;
  }, {});
```

I'm not really sure, but I don't think I can just replace this with Object.fromEntries, can I?
If not, maybe it should try finding when we create a pair of k/v out of an object, and report only then?
Or update the error message to tell us to use `fromEntries` *and* another `Array` method (map or whatever) to make our modifications? Because here it looks like it tells me to only use fromEntries, but I don't see how I can 😅

EDIT: I just realised here, my `stores` is a [`Collection`](https://discord.js.org/#/docs/collection/main/class/Collection) (a Map on steroid, and it has a reduce method). So maybe it's just that the rule shouldn't apply here and I have to disable it.

#### Unrelated

It would be great to enable Github Discussions, as I'm not sure this is a bug, so a Discussion would have been more appropriate IMO :smile:

**Comment by fregante** [comment]
```is
const categories = {};
for (const val of stores) {
    const category = getCategory(val);
	categories[category] = categories[category] ?? [];
	categories[category].push(val)
}
```

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-reduce.md

**Comment by papb** [comment]
Or, a little closer to your original code structure:

```js
const categories = {};
for (const val of stores) {
    const category = getCategory(val);
    if (categories[category]) {
        categories[category].push(val);
    } else {
        categories[category] = [val];
    }
}
```

**Comment by papb** [comment]
That said, I just noticed that `prefer-object-from-entries` is basically complaining of any `reduce` call with `{}` as second argument. However, clearly only a very small subset of these calls are actually replaceable by `Object.fromEntries()`. I'd call this a bug.

```js
const someFn = require('./whatever');
const obj = arr.reduce(someFn, {}); // Prefer `Object.fromEntries()` over `Array#reduce()`.
```

Some options that come to my mind:

1. Change the rule to report only the cases it is 100% sure (i.e. the currently autofixable cases)
2. Change the rule to be a warning rather than error and change the message to something softer, such as:

    > This may be replaceable with `Object.fromEntries` which is more readable

    * Note: I don't know if it's possible in ESLint to make a rule that will sometimes be a warning and sometimes an error - if yes, then it could still be an error for the autofixable cases, while being a warning to the other cases.

I prefer option 1, because I don't like the idea of the linter making bold assumptions over `.reduce(whatever, {})`.

That said, I am a fan of `no-array-reduce` anyway, so this doesn't really impact me at all.

**Comment by fregante** [comment]
You may be right on this one. I see an overlap with no-array-reduce and an impossible suggestion (because not all `reduce`s can be turned into an Object.fromEntries)


Maybe the scope of the rule should be *reduced.*

**Comment by fisker** [comment]
Make sense to me, let's remove those cases that we don't know what's in the callback, we have `no-array-reduce` anyway.
